dgc/Sign1CoseMessage.cs

```diff
@@ -0,0 +1,279 @@
+                protectedMap[HeaderKey_Alg] = Alg_ES256;
+                signer = SignerUtilities.GetSigner("SHA-256withECDSA");
+                signLenght = 32;
+            }
+            else if (alg == HCertSupportedAlgorithm.PS256)
+            {
+                RegisteredAlgorithm = HCertSupportedAlgorithm.PS256;
+                protectedMap[HeaderKey_Alg] = Alg_PS256;
+
+                signer = SignerUtilities.GetSigner("SHA256withRSA/PSS");
+            }
+            else
+            {
+                throw new NotSupportedException("Algorithm not supported");
+            }
+
+            signer.Init(true, keypair.Private);
+            var cborArray = CBORObject.NewArray();
+            cborArray.Add(ContextSignature1);
+            cborArray.Add(protectedMap.EncodeToBytes());
+            cborArray.Add(new byte[0]);
+            cborArray.Add(Content);
+
+            var bytesToSign = cborArray.EncodeToBytes();
+            signer.BlockUpdate(bytesToSign, 0, bytesToSign.Length);
+
+            var signature = signer.GenerateSignature();
+            if (signLenght > 0)
+            {
+                signature = ConvertDerToConcat(signature, signLenght);
+            }
+
+            Signature = signature;
+
+            KID = keyid;
+        }
+
+        public bool VerifySignature(AsymmetricKeyParameter pubKey)
+        {
+            ISigner signer;
+            var signature = Signature;
+            if (pubKey is Org.BouncyCastle.Crypto.Parameters.ECKeyParameters)
+            {
+                signer = SignerUtilities.GetSigner("SHA-256withECDSA");
+                signature = ConvertConcatToDer(Signature);
+            }
+            else if (pubKey is Org.BouncyCastle.Crypto.Parameters.RsaKeyParameters)
+            {
+                signer = SignerUtilities.GetSigner("SHA256withRSA/PSS");
+            }
+            else
+            {
+                throw new ArgumentException("Algorithm not supported");
+            }
+ 
+            signer.Init(false, pubKey);
+            
+            var cborArray = CBORObject.NewArray();
+            cborArray.Add(ContextSignature1);
+            cborArray.Add(ProtectedMap.EncodeToBytes());
+            cborArray.Add(new byte[0]); // no externaldata
+            cborArray.Add(Content);
+
+            var bytesToSign = cborArray.EncodeToBytes();
+            // if ec then encode
+            signer.BlockUpdate(bytesToSign, 0, bytesToSign.Length);
+
+            return signer.VerifySignature(signature);
+        }
+
+        private static byte[] ConvertConcatToDer(byte[] concat)
+        {
+            int len = concat.Length / 2;
+
+            byte[] r = new byte[len];
+            Array.Copy(concat, 0, r, 0, len);
+            r = UnsignedInteger(r);
+
+            byte[] s = new byte[len];
+            Array.Copy(concat, len, s, 0, len);
+            s = UnsignedInteger(s);
+
+            var x = new List<byte[]>();
+            x.Add(new byte[] { 0x30 });
```

Comment:
magic number 0x30? Appreciate is a "known value" -> but then in that case it should be easy to come up with a name for a const value to represent it ;)